src/Block/MediaBlockService.php

```diff
@@ -205,11 +206,17 @@ private function getMediaBuilder(): FormBuilderInterface
             'edit' => 'list',
         ]);
 
-        return $this->mediaAdmin->getFormBuilder()->create('mediaId', ModelListType::class, [
+        $fieldDescription->setAssociationAdmin($this->mediaAdmin);
+
+        $formBuilder = $this->mediaAdmin->getFormBuilder()->create('mediaId', ModelListType::class, [
             'sonata_field_description' => $fieldDescription,
             'class' => $this->mediaAdmin->getClass(),
             'model_manager' => $this->mediaAdmin->getModelManager(),
             'label' => 'form.label_media',
         ]);
+
+        // Add ModelTransformer ?
```

Comment:
Yes we should add the model transformer, in this class there is a pre persist and pre update methods that are never called and should be removed in favour of the tranformer.

Reply:
Sorry this waiting time. 
Okay i see, but actually it was the ```getFormBuilder()``` method who give me the error, so i cant add the transformer before